### PR TITLE
Update Intel libraries (MKL, TBB, IPP). Remove parallelSTL / oneDPL.

### LINKED ISSUE #6544: Cannot compile on Ubuntu 23.10
### Checklist

- [X] I have searched for [similar issues](https://github.com/isl-org/Open3D/issues).
- [X] For Python issues, I have tested with the [latest development wheel](http://www.open3d.org/docs/latest/getting_started.html#development-version-pip).
- [X] I have checked the [release documentation](http://www.open3d.org/docs/release/) and the [latest documentation](http://www.open3d.org/docs/latest/) (for `main` branch).

#### I first cloned Open3D by:
```
git clone https://github.com/isl-org/Open3D.git
cd Open3D
```

#### Then, I build Open3D (on Ubuntu 23.10) with:
```
mkdir build
cd build
cmake -DBUILD_SHARED_LIBS=ON -DCMAKE_INSTALL_PREFIX=${HOME}/open3d_install ..
make install -j 12
```

### Error message

```shell
In file included from /usr/include/c++/13/bits/specfun.h:43,
                 from /usr/include/c++/13/cmath:3699,
                 from /home/marius/projects/ciw/Open3D/build/assimp/src/ext_assimp/include/assimp/vector2.h:53,
                 from /home/marius/projects/ciw/Open3D/build/assimp/src/ext_assimp/include/assimp/types.h:64,
                 from /home/marius/projects/ciw/Open3D/build/assimp/src/ext_assimp/include/assimp/IOStream.hpp:53,
                 from /home/marius/projects/ciw/Open3D/build/assimp/src/ext_assimp/include/assimp/DefaultIOStream.h:55,
                 from /home/marius/projects/ciw/Open3D/build/assimp/src/ext_assimp/include/assimp/Exceptional.h:49,
                 from /home/marius/projects/ciw/Open3D/build/assimp/src/ext_assimp/include/assimp/BaseImporter.h:52,
                 from /home/marius/projects/ciw/Open3D/build/assimp/src/ext_assimp/code/AssetLib/MDL/MDLLoader.h:50,
                 from /home/marius/projects/ciw/Open3D/build/assimp/src/ext_assimp/code/AssetLib/MDL/MDLLoader.cpp:51:
In static member function ‘static _Up* std::__copy_move<_IsMove, true, std::random_access_iterator_tag>::__copy_m(_Tp*, _Tp*, _Up*) [with _Tp = long unsigned int; _Up = long unsigned int; bool _IsMove = false]’,
    inlined from ‘_OI std::__copy_move_a2(_II, _II, _OI) [with bool _IsMove = false; _II = long unsigned int*; _OI = long unsigned int*]’ at /usr/include/c++/13/bits/stl_algobase.h:506:30,
    inlined from ‘_OI std::__copy_move_a1(_II, _II, _OI) [with bool _IsMove = false; _II = long unsigned int*; _OI = long unsigned int*]’ at /usr/include/c++/13/bits/stl_algobase.h:533:42,
    inlined from ‘_OI std::__copy_move_a(_II, _II, _OI) [with bool _IsMove = false; _II = long unsigned int*; _OI = long unsigned int*]’ at /usr/include/c++/13/bits/stl_algobase.h:540:31,
    inlined from ‘_OI std::copy(_II, _II, _OI) [with _II = long unsigned int*; _OI = long unsigned int*]’ at /usr/include/c++/13/bits/stl_algobase.h:633:7,
    inlined from ‘std::vector<bool, _Alloc>::iterator std::vector<bool, _Alloc>::_M_copy_aligned(const_iterator, const_iterator, iterator) [with _Alloc = std::allocator<bool>]’ at /usr/include/c++/13/bits/stl_bvector.h:1306:28,
    inlined from ‘void std::vector<bool, _Alloc>::_M_reallocate(size_type) [with _Alloc = std::allocator<bool>]’ at /usr/include/c++/13/bits/vector.tcc:851:40,
    inlined from ‘void std::vector<bool, _Alloc>::reserve(size_type) [with _Alloc = std::allocator<bool>]’ at /usr/include/c++/13/bits/stl_bvector.h:1094:17,
    inlined from ‘Assimp::MDL::IntSharedData_MDL7::IntSharedData_MDL7()’ at /home/marius/projects/ciw/Open3D/build/assimp/src/ext_assimp/code/AssetLib/MDL/MDLFileData.h:879:32,
    inlined from ‘void Assimp::MDLImporter::InternReadFile_3DGS_MDL7()’ at /home/marius/projects/ciw/Open3D/build/assimp/src/ext_assimp/code/AssetLib/MDL/MDLLoader.cpp:1351:29:
/usr/include/c++/13/bits/stl_algobase.h:437:30: error: ‘void* __builtin_memmove(void*, const void*, long unsigned int)’ forming offset 8 is out of the bounds [0, 8] [-Werror=array-bounds=]
  437 |             __builtin_memmove(__result, __first, sizeof(_Tp) * _Num);
      |             ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
make[5]: *** [code/CMakeFiles/assimp.dir/build.make:1504: code/CMakeFiles/assimp.dir/AssetLib/MDL/MDLLoader.cpp.o] Error 1
make[5]: *** Waiting for unfinished jobs....
make[4]: *** [CMakeFiles/Makefile2:169: code/CMakeFiles/assimp.dir/all] Error 2
make[3]: *** [Makefile:136: all] Error 2
make[2]: *** [CMakeFiles/ext_assimp.dir/build.make:86: assimp/src/ext_assimp-stamp/ext_assimp-build] Error 2
make[1]: *** [CMakeFiles/Makefile2:1011: CMakeFiles/ext_assimp.dir/all] Error 2
```

### Open3D, Python and System information

```markdown
- Operating system: Ubuntu 23.10
- Python version: Python 3.11.6 (main, Oct  8 2023, 05:06:43) [GCC 13.2.0] / output from `import sys; print(sys.version)`
- System architecture: x86
- Is this a remote workstation?: no
- How did you install Open3D?: build from source
- Compiler version (if built from source): gcc 13.2.0 / clang 16.0.6
```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by ssheorey**
1. Add VERSION (API version - major.minor.patch) and SOVERSION (ABI version major.minor) to libOpen3D
2. Preparing for OpenMP -> TBB conversion. TBB 2021.12.
3. Preparing for Intel GPU offload with new MKL (2024.1 for Win, Linux and 2023.2.2 for macOS).
4. Additional computer vision / image processing functions (e.g. optical flow) from new IPP (2021.11 for Win/Linux and 2021.9.1 for macOS)). No GPU support yet, but get automatic TBB based multi-threading inside image processing operations.
5. Switch to C++17.
6. TBB is now linked dynamically. TBB static linking is not supported and may cause over-subscription / crashes, especially in Python. TBB DSOs are now distributed with Open3D.

## Type

-   [x] Bug fix (non-breaking change which fixes an issue): Fixes #6544 (partially)
-   [ ] New feature (non-breaking change which adds functionality). Resolves #
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected) Resolves #

## Checklist:

-   [x] I have run `python util/check_style.py --apply` to apply Open3D **code style**
    to my code.
-   [x] This PR changes Open3D behavior or adds new functionality.
    -   [ ] Both C++ (Doxygen) and Python (Sphinx / Google style) **documentation** is
        updated accordingly.
    -   [ ] I have added or updated C++ and / or Python **unit tests** OR included **test
        results** (e.g. screenshots or numbers) here.
-   [x] I will follow up and update the code if CI fails.
    
-   [ ] For fork PRs, I have selected **Allow edits from maintainers**.

### CODE REVIEW

**Comment by ssheorey** [comment]
Ready to review....
macOS failure is due to download error. Earlier runs passed.

File: python/open3d/__init__.py [hunk 1]
```diff
@@ -31,13 +35,16 @@ def load_cdll(path):
...
-    if sys.platform == 'win32' and sys.version_info >= (3, 8):
+    if sys.platform == "win32" and sys.version_info >= (3, 8):
         # https://stackoverflow.com/a/64472088/1255535
         return CDLL(str(path), winmode=0)
     else:
         return CDLL(str(path))
 
 
+if sys.platform == "win32":  # Unix: Use rpath to find libraries
+    _win32_dll_dir = os.add_dll_directory(str(Path(__file__).parent))
```
Comment (benjaminum) [inline python/open3d/__init__.py:46]: Can this have side effects for other libs?

File: 3rdparty/find_dependencies.cmake [hunk 2]
```diff
@@ -265,9 +265,13 @@ endfunction()
 #    LIBRARIES
 #        the expected library variable names to be found in <pkg>.
 #        If <pkg> also defines targets, use them instead and pass them via TARGETS option.
+#    PATHS
```
Comment (benjaminum) [inline 3rdparty/find_dependencies.cmake:268]: Minimal docstrings needed

**Review by benjaminum — CHANGES_REQUESTED** [review]
Just some minor questions from my side.

Is there anything that needs special attention?

File: cmake/Open3DSetGlobalProperties.cmake [hunk 3]
```diff
@@ -25,12 +25,24 @@ function(open3d_enable_strip target)
...
 
+# RPATH handling (for TBB DSO). Check current folder, one folder above and the lib sibling folder 
+set(CMAKE_BUILD_RPATH_USE_ORIGIN ON)
+if (APPLE)
+# Add options to cover the various ways in which open3d shaed lib or apps can be installed wrt TBB DSO
+    set(CMAKE_INSTALL_RPATH "@loader_path;@loader_path/../;@loader_path/../lib/")
+# pybind with open3d shared lib is copied, not cmake-installed, so we need to add .. to build rpath 
+    set(CMAKE_BUILD_RPATH "@loader_path/../")   
+elseif(UNIX)
+    set(CMAKE_INSTALL_RPATH "$ORIGIN;$ORIGIN/../;$ORIGIN/../lib/")
```
[the remark this answers is not in the document]
Comment (ssheorey) [reply cmake/Open3DSetGlobalProperties.cmake:36]: Yes, that is for the binary packages (both Linux and macOS have that folder structure)

File: python/open3d/__init__.py (hunk 1 above)
Comment (ssheorey) [reply python/open3d/__init__.py:46]: No. The `_win32_dll_dir.close()` at the end of `__init__.py` reverts this change. Modules imported after Open3D will not see this dll directory.

**Review by benjaminum — APPROVED** [review]
lgtm

**Pull request merged**